### ISSUE TITLE
epic: voice rewrite

### ISSUE THREAD

**Issue by insertish** [revoltchat/backend#313]
This is the tracking issue for the voice rewrite.

**Comment by nlutterman** [comment]
Is there a need for assistance with this? Is this work on hold because there isn't enough bandwidth?

I'd be willing to lend a hand if that's the case.

**Comment by Zomatree** [comment]
Its in development, you can see the progress at https://github.com/revoltchat/backend/pull/318

**Comment by MarcheloTchelo** [comment]
I want help i need know and learn to help to accel this process because i some happy with revolt and 75% migrate when this function start BETA ALL for clients i migrate discord to REVOLT 100%

**Comment by bignegro** [comment]
nope its on hold for i guess forever its cooked

**Comment by nlutterman** [comment]
It's a pretty big project. I've been meaning to try and find the time to contribute. Hopefully I can soon.

**Comment by maxi1134** [comment]
This will be a great addition!

Especially now that Oligarchs control the traditional social networks.

Gotta go private.

**Comment by Tech-User42** [comment]
It would be so nice if we could test even a buggy version, that could take me closer to my dream of leaving Discord

**Comment by EntityinArray** [comment]
```
(non-functional) Users are automatically disconnected from voice calls when they are idle.
```
I think this is an anti-feature. A much more elegant approach would be to keep user in the voice channel, but make voice connection dormant (stop sending any audio data) . Then start sending audio again as soon as sound is detected.
Even then, I think this overengineering is not needed at all. Opus audio requires little bandwith and already optimizes silence.

Users can judge whether user is AFK by presence

**Comment by 7b9e1e** [comment]
the lack of video streaming is the main thing preventing my friends from moving to revolt, with discord's upcoming AI age verification we came back to revolt to check in again, still no progress on this?

**Comment by Zomatree** [comment]
> the lack of video streaming is the main thing preventing my friends from moving to revolt, with discord's upcoming AI age verification we came back to revolt to check in again, still no progress on this? 

The work is being done here https://github.com/revoltchat/backend/pull/414

**Comment by 7b9e1e** [comment]
> > the lack of video streaming is the main thing preventing my friends from moving to revolt, with discord's upcoming AI age verification we came back to revolt to check in again, still no progress on this?
> 
> The work is being done here [#414](https://github.com/revoltchat/backend/pull/414)

thanks for replying, i wish you luck in implementing more features!

**Comment by Eddudos** [comment]
What is the latest situation on solfhosted voice channels?
I saw that #414 is merged, so it should be replaced in selfhosted version too?

**Comment by insertish** [comment]
> What is the latest situation on solfhosted voice channels? I saw that [#414](https://github.com/stoatchat/stoatchat/pull/414) is merged, so it should be replaced in selfhosted version too?

Yes! Tracking issue: https://github.com/stoatchat/self-hosted/issues/176
It also necessarily needs the new client, since we didn't backport voice features.

**Comment by EgeGurkan** [comment]
Just add screen share so we stop using discord come on!

**Comment by DashieTM** [comment]
> Just add screen share so we stop using discord come on!

Open a PR then, this isn't a paid service.

**Comment by pparaxan** [comment]
"Talk is cheap, send patches"

**Comment by amycatgirl** [comment]
> Just add screen share so we stop using discord come on!


Video functionality is already implemented, but it isn't ready to be launched yet because Stoat needs to prepare their servers to handle said feature, as it's resource intensive.

If you are running a self-hosted instance (not using docker), you are already able to enable it for your instance by editing the `Revolt.toml` file. (Documentation for all the available options: https://github.com/stoatchat/stoatchat/blob/main/crates/core/config/Revolt.toml)

**Comment by EgeGurkan** [comment]
found nerimity nevermind

**Comment by ROODAY** [comment]
Just curious, what is the current status of voice channels? I just followed the self-hosting guide to get an instance up, and then trying to create a voice channel just makes a text channel. But the linked PR at the top https://github.com/stoatchat/stoatchat/pull/414 seems to be merged already?

**Comment by amycatgirl** [comment]
For self-hosted, go to https://github.com/stoatchat/self-hosted/issues/176

**Comment by javif89** [comment]
> Just curious, what is the current status of voice channels? I just followed the self-hosting guide to get an instance up, and then trying to create a voice channel just makes a text channel. But the linked PR at the top [#414](https://github.com/stoatchat/stoatchat/pull/414) seems to be merged already?

@ROODAY 

This thread is very helpful: https://github.com/stoatchat/self-hosted/issues/176

Managed to get it working today and put my findings down here:

https://github.com/javif89/stoat-selfhost

If there's something incorrect in the guide or things change, feel free to make a PR or just link a new guide. I'm sure a lot of people are looking for answers right now.

**Comment by ChrisBlaylock13** [comment]
I think it's safe to say that video streaming is the biggest wanted feature so that people can leave Discord, especially with what's going on there now. I hope this feature is finished out, and server upgraded soon. It's the only reason I haven't got my group to switch yet.

**Comment by kuolemaaa** [comment]
> I think it's safe to say that video streaming is the biggest wanted feature so that people can leave Discord, especially with what's going on there now. I hope this feature is finished out, and server upgraded soon. It's the only reason I haven't got my group to switch yet.

nobody owes anything and nothing is owed. send patches

**Comment by IngwiePhoenix** [comment]
> send patches

Heh, ffmpeg approach much? :p Already learning the ins and outs of LiveKit. Also trying to find a good way to help out on the selfhosting front. Hoping to find a good way to contribute sooner rather than later. Two communities I have want to migrate, so theres some pressure. A healthy amount, to be fair.

**Comment by kuolemaaa** [comment]
@IngwiePhoenix just to be clear: i am not affiliated with this project. i am interested in it too and I’m studying the codebase. having said that we will leave this place to more related conversation from now on.

**Comment by EgeGurkan** [comment]
a project that author says nothing is promised or nothing is owed is just doomed to die, in a place everyone is rude and no one tries to help nothing will grow. switch to other projects, don't waste time here

**Comment by IAmTomahawkx** [comment]
This is accruing too much conversation that is not related to the actual development of the feature, as such it's been locked.
